readthedocs/projects/models.py

```diff
@@ -1570,6 +1570,43 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
         super().save(*args, **kwargs)
 
 
+class HTTPHeader(TimeStampedModel, models.Model):
+
+    """
+    Define a HTTP header for a user Domain.
+
+    All the HTTPHeader(s) associated with the domain are added in the response
+    from El Proxito.
+
+    NOTE: the available headers are hardcoded in the NGINX configuration for
+    now (see ``dockerfile/nginx/proxito.conf``) until we figure it out a way to
+    expose them all without hardcoding them.
```

Comment:
I realize there's a desire to be generic but setting arbitrary headers is potentially dangerous. I prefer this set list of headers although we also could probably set any header that starts `X-` as well.